### PR TITLE
feat: update permission rule when role updated

### PR DESCRIPTION
**Pull request by notdu**
Continue PR: https://github.com/casdoor/casdoor/pull/1153

### CODE REVIEW

**Comment by notdu** [comment]
continue the PR #1153

**Comment by hsluoyz** [comment]
@leo220yuyaodog plz review

File: object/permission_enforcer.go [hunk 1]
```diff
@@ -72,86 +72,118 @@ m = g(r.sub, p.sub) && r.obj == p.obj && r.act == p.act`
...
 		roleObj := GetRole(role)
+		if roleObj == nil {
+			continue
+		}
+
 		for _, subUser := range roleObj.Users {
 			if domainExist {
 				for _, domain := range permission.Domains {
-					groupingPolicies = append(groupingPolicies, []string{subUser, domain, role, "", "", permissionId})
+					groupingPolicies = append(groupingPolicies, []string{subUser, domain, role, "", "", permissionID})
```
Comment (hsluoyz) [inline object/permission_enforcer.go:129]: Use `permissionId`

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Review by hsluoyz — APPROVED** [review]

**Comment by hsluoyz** [comment]
@notdu plz fix conflicts

image

**Comment by leo220yuyaodog** [comment]
Need to do some compatiable, like these pr #1478 
If use rbac model, the table permission_rule will has g type policy 
image
But if other permission use models that do not have `[role_definition]`, it will cause when init enforce.
image

See the image, the model don't has g type
image

**Comment by notdu** [comment]
Hi @hsluoyz, @leo220yuyaodog,
In Casbin, we don't check if the `assertion` is exist or not when `LoadPolicyArray`, is it ok if i create a PR to fix it.
The problem above will be solved.
Screenshot 2023-01-15 at 6 05 21 PM

**Comment by hsluoyz** [comment]
@notdu if the Casbin model is not a RBAC enabled model, then all `g` rules should be ignored (not loaded into the enforcer)

**Comment by notdu** [comment]
@hsluoyz by adding 
` if assertion := m[sec][key]; assertion == nil { 
	return nil
}`
to `LoadPolicyArray` 
i think we can handle it, or anything wrong with my solution?

**Comment by hsluoyz** [comment]
@notdu it seems to be OK. Can you update the PR so I can see the whole code?

**Comment by notdu** [comment]
@hsluoyz Nothing change on this PR, i'll create a PR on casbin, after that PR is merged, i'll update this PR

**Comment by hsluoyz** [comment]
@notdu the whole change should happen in this repo. Casbin side is not needed to be changed.

**Comment by leo220yuyaodog** [comment]
> @hsluoyz by adding ` if assertion := m[sec][key]; assertion == nil { return nil }` to `LoadPolicyArray` i think we can handle it, or anything wrong with my solution?

https://github.com/casdoor/casdoor/blob/d0741e37057283c67e8f13e4db44b0609f32e8ad/object/permission_enforcer.go#L53
Can add the condition to filter policy of the adapter according to the model text before use the adapter to init the enforce.
```go
if !strings.Contains(permissionModel.ModelText,"[role_definition]"){
	adapter.LoadFilteredPolicy(m, xormadapter.Filter{Ptype: []string{"p"}})
}
```

**Comment by notdu** [comment]
@leo220yuyaodog, @hsluoyz 
plz review my new updates

**Comment by leo220yuyaodog** [comment]
> @leo220yuyaodog, @hsluoyz plz review my new updates

lgtm

**Pull request merged**